src/main/java/org/haffson/adventofcode/days/day01/Day01.java

```diff
@@ -20,6 +21,13 @@
 
     private final int[] numbers;
 
+    Day01() {
+        this.problemStatus = new HashMap<>();
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.SOLVED);
+        this.numbers = new int[]{};
+    }
+
```

Comment:
Are you using this constructor, so that you don't have to use the `FileReaders` in the test? In this case, you could use the other constructor and provide a `FilesReaders` mock. Then, you don't need this extra constructor:

You could try something like this:
````
import org.haffson.adventofcode.days.Days;
import org.haffson.adventofcode.days.day01.Day01;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
public class FileReadersTest {

    @Test
    void test_x(){
        int[] values = new int[]{979, 366, 675};

        FileReaders fileReaders = mock(FileReaders.class);
        when(fileReaders.readFileIntoArrayOfIntegers(anyString()))
                .thenReturn(values);

        int expectedResult = 241861950;

        Days day01 = new Day01(fileReaders, "path");

        int actualResult = day01.multiplyListEntries(values);

        assertThat(actualResult).isEqualTo(expectedResult);
    }

}

````

You will also need the mockito dependency:
````
    testImplementation 'org.junit.jupiter:junit-jupiter:5.6.0'
    testImplementation 'org.mockito:mockito-junit-jupiter:3.3.3'
````

Reply:
The problematic thing about this is that you never know what to expect from `numbers`. Does it contain values or not? You end up needing quite some checks in the methods using that variable because you have two constructors that set `numbers` differently. It is better to have just one specific and reliable behavior.

Also, your methods don't use the class variable `numbers` but one that is passed to them via the method parameter. If these methods should be used as util methods, this is fine. In your case, you could just the class variable.